sdk/src/wallet/mod.rs

```diff
@@ -225,6 +226,31 @@ pub mod fs {
     }
 }
 
+/// Generate a new secret key.
+pub fn gen_secret_key(
+    scheme: SchemeType,
+    csprng: &mut (impl CryptoRng + RngCore),
+) -> common::SecretKey {
+    match scheme {
+        SchemeType::Ed25519 => ed25519::SigScheme::generate(csprng).try_to_sk(),
+        SchemeType::Secp256k1 => {
+            secp256k1::SigScheme::generate(csprng).try_to_sk()
+        }
+        SchemeType::Common => common::SigScheme::generate(csprng).try_to_sk(),
```

Comment:
What is this?

Reply:
This function used to be called `gen_sk_rng` and lived in`sdk/src/wallet/store.rs`. Each implementation of signing scheme has to be identified with a `SchemeType` (`SchemeType::Secp256k1`, `SchemeType::Ed25519`, ...) (because of `SigScheme` trait requirements). Since we needed a signing scheme that is not fixed at compile-time, we ended up making `common::PublicKey`, `common::SecretKey`, `common::Signature` and identifying them with `SchemeType::Common`.

Since `common::SigScheme::generate(csprng)` panics, the entire branch `SchemeType::Common => common::SigScheme::generate(csprng).try_to_sk(),` panics. We could probably make that more clearer.